llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1213,16 +1216,21 @@ translateSpirvGlobalUses(LoadInst *LI, StringRef SpirvGlobalName,
     InstsToErase.push_back(LI);
     return;
   }
-
+  bool isLIErased = false;
   // Only loads from _vector_ SPIRV globals reach here now. Their users are
   // expected to be ExtractElementInst or TruncInst only, and they are replaced
   // in this loop. When loads from _scalar_ SPIRV globals are handled here as
   // well, the users will not be replaced by new instructions, but the GenX call
   // replacing the original load 'LI' should be inserted before each user.
   for (User *LU : LI->users()) {
-    assert(
-        (isa<ExtractElementInst>(LU) || isa<TruncInst>(LU)) &&
-        "SPIRV global users should be either ExtractElementInst or TruncInst");
+    // Ignore pointer type uses
+    if (LU->getType()->getTypeID() == llvm::Type::TypeID::PointerTyID)
```

Comment:
I think this is too generic. You basically need to distinguish the
`%0 = load i64, i64 addrspace(1)* getelementptr (<3 x i64>, <3 x i64> addrspace(1)* @__spirv_BuiltInGlobalInvocationId, i64 0, i64 0), align 32`
case from others. This case above performs thread_id calculation in one instruction basically, unlike the other two variants, where load is followed by element extraction or truncation.